multisiteauth/settings.py

```diff
@@ -1,8 +1,22 @@
 from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+from django.utils.module_loading import import_string
+
 
 HTTP_AUTH_ENABLED = getattr(settings, 'BASIC_HTTP_AUTH_ENABLED', False)
 HTTP_AUTH_GENERAL_USERNAME = getattr(settings, 'BASIC_HTTP_AUTH_GENERAL_USERNAME', '')
 HTTP_AUTH_GENERAL_PASS = getattr(settings, 'BASIC_HTTP_AUTH_GENERAL_PASS', '')
 HTTP_AUTH_ALLOW_ADMIN = getattr(settings, 'BASIC_HTTP_AUTH_ALLOW_ADMIN', True)
 HTTP_AUTH_REALM = getattr(settings, 'BASIC_HTTP_AUTH_REALM', '')
 HTTP_AUTH_URL_EXCEPTIONS = getattr(settings, 'BASIC_HTTP_AUTH_URL_EXCEPTIONS', [])
+
+HTTP_AUTH_GET_CURRENT_SITE = getattr(
+    settings,
+    'BASIC_HTTP_AUTH_GET_CURRENT_SITE',
+    'django.contrib.sites.shortcuts.get_current_site')
+
+try:
+    get_current_site = import_string(HTTP_AUTH_GET_CURRENT_SITE)
+except ImportError as exc:
+    err_msg = "Cannot import {}! This is used to get the current site."
```

Comment:
If this fails because the user configured the setting wrong he could see:
ImportError: a_x doesn't not look like a module path.
ImportError: Module a does not define a b. attribute/class.
or how it is now:
ImproperlyConfigured: "Cannot import a_x! This is used to get the current site.".

While I agree that this does not give the user information about what part of the string is wrong, it also looks like an application error, not a configuration one. I take if more as "Ah, an ImportError, I know to handle this problem and inform the user, even if hiding the  original stack trace is usually bad."

We can go either way with this. I forgot to add `.format(..` anyway.